src/bank.rs

```diff
@@ -1886,4 +1940,90 @@ mod tests {
         // hash3 is not in the q
         assert_eq!(Bank::check_entry_id_age(&q, hash3, 3), false);
     }
+    #[test]
+    fn test_first_err() {
+        assert_eq!(Bank::first_err(&[Ok(())]), Ok(()));
+        assert_eq!(
+            Bank::first_err(&[Ok(()), Err(BankError::DuplicateSignature)]),
+            Err(BankError::DuplicateSignature)
+        );
+        assert_eq!(
+            Bank::first_err(&[
+                Ok(()),
+                Err(BankError::DuplicateSignature),
+                Err(BankError::AccountInUse)
+            ]),
+            Err(BankError::DuplicateSignature)
+        );
+        assert_eq!(
+            Bank::first_err(&[
+                Ok(()),
+                Err(BankError::AccountInUse),
+                Err(BankError::DuplicateSignature)
+            ]),
+            Err(BankError::AccountInUse)
+        );
+        assert_eq!(
+            Bank::first_err(&[
+                Err(BankError::AccountInUse),
+                Ok(()),
+                Err(BankError::DuplicateSignature)
+            ]),
+            Err(BankError::AccountInUse)
+        );
+    }
+    #[test]
+    fn test_par_process_entries() {
```

Comment:
Can you split this into 4 tests? I don't mind the duplication of those first 4 lines. It's easier to grok than figuring out where one test depends on data or side-effects of the previous.